src/main/java/org/apache/maven/plugins/install/InstallFileMojo.java

```diff
@@ -218,7 +218,7 @@ public void execute() throws MojoExecutionException, MojoFailureException {
             ArtifactType artifactType =
                     repositorySystemSession.getArtifactTypeRegistry().get(packaging);
             if (artifactType != null
-                    && StringUtils.isEmpty(classifier)
+                    && (classifier == null || classifier.isEmpty())
                     && !StringUtils.isEmpty(artifactType.getClassifier())) {
```

Comment:
Noticed it missed one in line 222

Reply:
That's surprising! Usually when that happens there's some missing type attribution; I'll do a final sweep to see if it pops up again.

Reply:
Looking at this again: the line was not missed, but intentionally left out in the recipe, as we can't be certain that the argument method invocation is safe to call twice; once for the null check, and again for the isEmpty call. Hence why I don't convert those automatically yet, even though in this case it's probably safe to do.